lib/src/git.rs

```diff
@@ -1239,6 +1359,30 @@ pub struct GitFetchStats {
 }
 
 #[tracing::instrument(skip(mut_repo, git_repo, callbacks))]
+pub fn clone(
+    mut_repo: &mut MutableRepo,
+    git_repo: &git2::Repository,
+    remote_name: &str,
+    branch_names: &[StringPattern],
+    callbacks: RemoteCallbacks<'_>,
+    git_settings: &GitSettings,
+    depth: Option<NonZeroU32>,
+) -> Result<GitFetchStats, GitFetchError> {
+    let mut git_fetch = GitFetch::new(
+        mut_repo,
+        git_repo,
+        git_settings,
+        GitFetch::fetch_options(callbacks, depth),
+    );
+    let default_branch = git_fetch.fetch(branch_names, remote_name)?;
+    let import_stats = git_fetch.import_refs(branch_names, &[remote_name])?;
+    let stats = GitFetchStats {
+        default_branch,
+        import_stats,
+    };
+    Ok(stats)
+}
+
 pub fn fetch(
```

Comment:
Can you port `fetch()` within this patch so the new `GitFetch` iplementation is covered by tests?

Reply:
Done.